presto-main/src/main/java/com/facebook/presto/operator/SimplePagesHashStrategy.java

```diff
@@ -200,6 +200,16 @@ public boolean positionEqualsPosition(int leftBlockIndex, int leftPosition, int
     @Override
     public boolean applyFilterFunction(int leftBlockIndex, int leftPosition, int rightPosition, Block[] allRightBlocks)
     {
-        return filterFunction.get().filter(leftPosition, channelArrays.get(leftBlockIndex), rightPosition, allRightBlocks);
+        return filterFunction.get().filter(leftPosition, getLeftBlocks(leftBlockIndex), rightPosition, allRightBlocks);
+    }
+
+    private Block[] getLeftBlocks(int leftBlockIndex)
+    {
+        if (channelArrays.isEmpty()) {
+            return EMPTY_BLOCK_ARRAY;
```

Comment:
I am thinking of query like this one.

```sql
SELECT t2.a FROM t1 JOIN t2 ON t2.c > t2.d
```


This is dumb query. But technically (if join condition is not pushed down to filter which is not always possible) it does not use any channels denoting columns from table t1. 

And that is exact case we are fighting here.

See testcases added in https://github.com/prestodb/presto/pull/5403/commits/f1403edef98bd3d8b3caef0477a9b853872824eb.